src/android/AckDatabase.java

```diff
@@ -2,22 +2,34 @@
 
 import android.content.Context;
 
+import androidx.annotation.NonNull;
 import androidx.room.Database;
 import androidx.room.Room;
 import androidx.room.RoomDatabase;
 import androidx.room.TypeConverters;
+import androidx.room.migration.Migration;
+import androidx.sqlite.db.SupportSQLiteDatabase;
 import androidx.work.Data;
 
-@Database(entities = {UploadEvent.class}, version = 5)
+@Database(entities = {UploadEvent.class}, version = 6)
 @TypeConverters(value = {Data.class})
 public abstract class AckDatabase extends RoomDatabase {
     private static AckDatabase instance;
 
-    public static AckDatabase getInstance(final Context context) {
+    static final Migration MIGRATION_5_6 = new Migration(5, 6) {
+        @Override
+        public void migrate(@NonNull SupportSQLiteDatabase database) {
+            // Add the new column to the existing table
+            database.execSQL("ALTER TABLE upload_events ADD COLUMN uploadDuration INTEGER NOT NULL DEFAULT 0");
+        }
```

Comment:
Instead of storing `uploadDuration`, why don't we store the `startTime` and `endTime`. The `uploadDuration` is calculated from these values. Normally it is better to save the values from which it calculated rather than the value itself. (Example saving the date of birth in a db rather than the age as age can be calculated with the dob)

If we save the the starting and ending times, we can then calculate the uploadDuration and return it alongside the ending time also to the JS and set our `uploadedAt` to the returned ending time value.

What do you think?

Reply:
Yes my suggestion was to calculate the `uploadDuration` on the plugin side itself and not calculate it on the JS side but to save the values from which we derive the `uploadDuration` in the db and return the `uploadDuration` AND the `uploadedAt` time also.

I gave the age as an example. Other examples can be keeping the `buyingPrice` and  `sellingPrice` then derive the `profit`from it althought the profit won't change. In this case the `uploadDuration` won't change but saving the start and end times gives more information than just the `uploadDuration`.